loader/readsource_test.go

```diff
@@ -51,7 +53,7 @@ func TestReadSourceSTDINCache(t *testing.T) {
 	var data = []byte(`test contents`)
 	var r = bytes.NewReader(data)
 	var fs = afero.NewMemMapFs()
-	sourceData, err := ReadSource("-", "/path/to/pwd",
+	sourceData, err := ReadSource(nil, "-", "/path/to/pwd",
```

Comment:
Couldn't this panic in practice if the loader tries logging? This test doesn't because it doesn't need it, and it's setup in such a way, but the fact that it can is a bit worrying.

Should we add a `nil` check and fallback to the global logger if a local one isn't provided?

Reply:
:facepalm: I've missed that - no I just should've given it a logger but apparently missed it when I decided that providing `nil` in test is bad idea ...

Reply:
fixed in 83fc72b

Reply:
Thanks, but shouldn't the fact that it's possible for this to panic if given a `nil` logger be a concern?

Reply:
literally we should never provide a `nil` logger, even in tests.

For me adding `* != nil` everywhere seems completely unneeded given that most things in k6 (including this) have 1-10 places where they are called (outside tests) and in all of those a `nil` logger is not a thing that should happen - it panicking (while awful) should never ever be possible in the first place, and masking it through using the global logger IMO is not what we want to do, we just want to always have defined the logger. 

Obviously if, with not checking everywhere, I could've made certain something isn't nil ever .. I would've, but (unfortunately?) this is not Rust ;) 

As I said in the commit message - this is called through 2 functions in 5 places in the actual (non-test) code and in all of those cases providing a `nil` logger seems completely unreasonable, and refactoring this into a loader struct which will just contain a `logger` at this point seems pretty pointless to me and won't even make any of the code smaller or easier to read (including the tests)